src/ProgressStack.php

```diff
@@ -4,60 +4,272 @@
 
 namespace Yiisoft\Yii\Bootstrap5;
 
-use Yiisoft\Arrays\ArrayHelper;
+use BackedEnum;
 use Yiisoft\Html\Html;
-
-use function array_map;
+use Yiisoft\Html\Tag\Div;
+use Yiisoft\Widget\Widget;
 
 /**
- * `ProgressStack` renders a bootstrap progress bar component.
+ * The progress stack component allows you to stack multiple progress bars on top of each other.
+ *
+ * For example,
+ *
+ * ```php
+ * echo ProgressStack::widget()
+ *    ->bars(
+ *        Progress::widget()
+ *            ->ariaLabel('Segment one')
+ *            ->id('segment-one')
+ *            ->percent(15),
+ *        Progress::widget()
+ *            ->ariaLabel('Segment two')
+ *            ->backGroundColor(BackgroundColor::SUCCESS)
+ *            ->id('segment-two')
+ *            ->percent(30),
+ *        Progress::widget()
+ *            ->ariaLabel('Segment three')
+ *            ->backGroundColor(BackgroundColor::INFO)
+ *            ->id('segment-three')
+ *            ->percent(20),
+ *    )
+ *    ->render();
  *
  * @see https://getbootstrap.com/docs/5.3/components/progress/#multiple-bars
  */
 final class ProgressStack extends Widget
 {
+    private const NAME = 'progress-stack';
+    private const PROGRESS_STACKED = 'progress-stacked';
+    private array $attributes = [];
+    private array $cssClasses = [];
+    private array $bars = [];
+    private bool|string $id = true;
+
     /**
-     * @var Progress[]
+     * Adds a set of attributes.
+     *
+     * @param array $attributes Attribute values indexed by attribute names. e.g. `['id' => 'my-id']`.
+     *
+     * @return self A new instance with the specified attributes added.
+     *
+     * Example usage:
+     * ```php
+     * $progressStack->addAttributes(['data-id' => '123']);
+     * ```
      */
-    private array $bars = [];
-    private array $options = [];
+    public function addAttributes(array $attributes): self
+    {
+        $new = clone $this;
+        $new->attributes = [...$this->attributes, ...$attributes];
+
+        return $new;
+    }
+
+    /**
+     * Adds one or more CSS classes to the existing classes.
+     *
+     * @param BackedEnum|string|null ...$class One or more CSS class names to add. Pass `null` to skip adding a class.
+     *
+     * @return self A new instance with the specified CSS classes added to existing ones.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     *
+     * Example usage:
+     * ```php
+     * $progressStack->addClass('custom-class', null, 'another-class', BackGroundColor::PRIMARY);
+     * ```
+     */
+    public function addClass(BackedEnum|string|null ...$class): self
+    {
+        $new = clone $this;
+        $new->cssClasses = [...$this->cssClasses, ...$class];
+
+        return $new;
+    }
+
+    /**
+     * Adds a CSS style.
+     *
+     * @param array|string $style The CSS style. If an array, the values will be separated by a space. If a string, it
```

Comment:
"If an array, the values will be separated by a space." is repeated 2 times.

Reply:
fixed.